SSA/Projects/InstCombine/update_alive_statements.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Released under Apache 2.0 license as described in the file LICENSE.
+
+import re
+
+def getProofs(lines):
+    proofs = []
+    
+    proof = []
+    for line in lines:
+        if line.startswith("-- Name"):
+            proofs.append(proof)
+            proof = []
+        proof.append(line)
+    proofs.append(proof)
+
+    return proofs[0], proofs[1:]
+
+def getStatement(preamble, proof):
+    f = open("AliveTest.lean", "w")
+
+    f.write("".join(preamble))
+    rewritten = []
+    name = ""
+    for line in proof:
+        if line.startswith("  apply"):
+            rewritten.append("  print_goal_as_error")
+            name = line.split(" ")[3][0:-1]
+        else:
+            rewritten.append(line)
+    f.write("".join(rewritten))
+
+    f.close()
+    
+    import subprocess
+    x = subprocess.run("(cd ../../../; lake build SSA.Projects.InstCombine.AliveTest)",
+        shell=True, capture_output=True)
+    
+    name = "\n\ntheorem " + name + " :\n"
+
+    if x.returncode == 0:
+        return ""
+    error = x.stderr.decode("utf-8")
+    error = error.split("\n")
+    stmt = name
+    stmt += re.sub(".*error: ", "", error[2])
+    if len(error) >= 6:
+        stmt += "\n" + error[3]
+    if len(error) >= 7:
+        stmt += "\n" + error[4]
+    if len(error) >= 8:
+        stmt += "\n" + error[5]
+    if len(error) >= 9:
+        stmt += "\n" + error[6]
+    if len(error) >= 10:
+        stmt += "\n" + error[7]
+    if len(error) >= 11:
+        stmt += "\n" + error[8]
+    print("\n")
+    stmt += " := by\n  simp_alive_undef\n  simp_alive_ops\n  simp_alive_case_bash\n  try alive_auto\n  try sorry"
+    print(stmt)
+    return stmt
+
+def filterProofs(preamble, proofs):
+    statements = []
+    for proof in proofs:
+        print("processing: " + proof[0])
+        stmt = getStatement(preamble, proof)
+        statements.append(stmt)
+
+    return statements
+
+def writeOutput(preamble, proofs, filename):
+    with open(filename, "w") as f:
+        f.write("".join(preamble))
+        for proof in proofs:
+            f.write("".join(proof))
+
+f = open("AliveAutoGenerated.lean", "r")
+lines = f.readlines()
+preamble, proofs = getProofs(lines)
+statements = filterProofs(preamble, proofs)
+# writeOutput(preamble, working_proofs, "AliveAutoGenerated.lean")
+preamble = """
+/-
+Released under Apache 2.0 license as described in the file LICENSE.
+-/
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.ForStd
+import SSA.Projects.InstCombine.ForMathlib
+import SSA.Projects.InstCombine.LLVM.Semantics
+import Std.Data.BitVec
+import Mathlib.Data.BitVec.Lemmas
+
+open LLVM
+open BitVec
+"""
+writeOutput(preamble, statements, "AliveStatements.lean")
```

Comment:
Add a `if __name__ == "__main_"` to allow for debugging by importing into a REPL. 

```suggestion
if __name__ == "__main__":
  f = open("AliveAutoGenerated.lean", "r")
  lines = f.readlines()
  preamble, proofs = getProofs(lines)
  statements = filterProofs(preamble, proofs)
  # writeOutput(preamble, working_proofs, "AliveAutoGenerated.lean")
  preamble = """
  /-
  Released under Apache 2.0 license as described in the file LICENSE.
  -/
  import SSA.Projects.InstCombine.TacticAuto
  import SSA.Projects.InstCombine.ForStd
  import SSA.Projects.InstCombine.ForMathlib
  import SSA.Projects.InstCombine.LLVM.Semantics
  import Std.Data.BitVec
  import Mathlib.Data.BitVec.Lemmas
  
  open LLVM
  open BitVec
  """
  writeOutput(preamble, statements, "AliveStatements.lean")
```